x-pack/plugins/profiling/server/lib/setup/fleet_policies.ts

```diff
@@ -0,0 +1,157 @@
+ * 2.0.
+ */
+
+import { merge, omit } from 'lodash';
+import { ElasticsearchClient } from '@kbn/core/server';
+import { getApmPolicy } from './get_apm_policy';
+import { ProfilingSetupOptions } from './types';
+
+async function createIngestAPIKey(esClient: ElasticsearchClient) {
+  const apiKeyResponse = await esClient.security.createApiKey({
+    name: 'profiling-manager',
+    role_descriptors: {
+      profiling_manager: {
+        indices: [
+          {
+            names: ['profiling-*', '.profiling-*'],
+            privileges: [
+              'read',
+              'create_doc',
+              'create',
+              'write',
+              'index',
+              'create_index',
+              'view_index_metadata',
+              'manage',
+            ],
+          },
+        ],
+        cluster: ['monitor'],
+      },
+    },
+  });
+
+  return atob(apiKeyResponse.encoded);
+}
+
+export async function validateApmPolicy({
+  soClient,
+  packagePolicyClient,
+}: ProfilingSetupOptions): Promise<boolean> {
+  const apmPolicy = await getApmPolicy({ packagePolicyClient, soClient });
+  return !!(apmPolicy && apmPolicy?.inputs[0].config?.['apm-server'].value?.profiling);
+}
+
+export async function updateApmPolicy({
+  client,
+  soClient,
+  packagePolicyClient,
+}: ProfilingSetupOptions) {
+  const esClient = client.getEsClient();
+  const apmPolicy = await getApmPolicy({ packagePolicyClient, soClient });
+
+  if (!apmPolicy) {
+    throw new Error(`Could not find APM policy`);
+  }
+
+  const apmPolicyApiKey = await createIngestAPIKey(esClient);
+
+  const profilingApmConfig = {
+    profiling: {
+      enabled: true,
+      elasticsearch: {
+        api_key: apmPolicyApiKey,
+      },
+      metrics: {
+        elasticsearch: {
+          hosts: ['https://1b6c02856ea642a6ac14499b01507233.us-east-2.aws.elastic-cloud.com:443'],
+          api_key: 'woq-IoMBRbbiEbPugtWW:_iBmc1PdSout7sf5FCkEpA',
+        },
+      },
+      keyvalue_retention: {
+        // 60 days
+        age: '1440h',
+        // 200 Gib
+        size_bytes: 200 * 1024 * 1024 * 1024,
+        execution_interval: '12h',
+      },
+    },
+  };
+
+  const modifiedPolicyInputs = apmPolicy.inputs.map((input) => {
+    return input.type === 'apm'
+      ? merge({}, input, { config: { 'apm-server': { value: profilingApmConfig } } })
+      : input;
+  });
+
+  await packagePolicyClient.update(soClient, esClient, apmPolicy.id, {
+    ...omit(apmPolicy, 'id', 'revision', 'updated_at', 'updated_by'),
+    inputs: modifiedPolicyInputs,
+  });
+}
+
+const CLOUD_AGENT_POLICY_ID = 'policy-elastic-agent-on-cloud';
+
+export async function createCollectorPackagePolicy({
+  client,
+  soClient,
+  packagePolicyClient,
+}: ProfilingSetupOptions) {
+  const packagePolicy = {
+    policy_id: CLOUD_AGENT_POLICY_ID,
+    enabled: true,
+    package: {
+      name: 'profiler_collector',
+      title: 'Universal Profiling Collector',
+      version: '8.9.0-preview',
```

Comment:
Is there a way to not hard code this version?

Reply:
I'm not sure exactly. As I understand it, the package policy versions used here are independent of the Elastic stack version and the current hardcoded versions are compatible with future stack versions until we update the package versions, so we shouldn't have to update often.

One strategy to remove the hardcoded versions could be to query the package registry for the most recent package version instead. However, there might be implications that I don't know about.

Perhaps there are alternative strategies.

@florianl @inge4pres Any thoughts on how to avoid hardcoding package versions in Kibana?

Reply:
you probably need to do something similar to this one https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/server/routes/fleet/get_apm_package_policy_definition.ts#L33-L36

Reply:
Thanks, I found and installed the latest versions for the collector and symbolizer package.